modules/server/src/network/utils/promise.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * Type guard to filter fulfilled Promises
+ */
+export const fulfilledPromiseFilter = <Result>(result: unknown): result is Result => {
+	return (
+		typeof result === 'object' &&
+		result !== null &&
+		'status' in result &&
+		result.status === 'fulfilled'
```

Comment:
thinking out loud to learn more, not a challenge or request for change:
curiously, "fulfilled" promises make sense, but I'd always thought of them as "resolved". it's a curious choice of synonym that I feel I may have missed became a standard somewhere.
was this "status == 'fulfilled'" a choice for the implementation here? or is it the output from a library? or simply a standard pattern I was not aware of?